bat/ssh.go

```diff
@@ -0,0 +1,85 @@
+//
+// Copyright (c) 2016 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package bat
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"golang.org/x/crypto/ssh"
+)
+
+//SSHClient containes details of the server to be connect
+type SSHClient struct {
+	ClientIP   string
+	ClientPort string
+	ClientUser string
+	ClientPass string
+}
+
+//CreateSSHClient creates a SSH session for the server specified
+//
+//Receive a struct with the details of the server that will be connected
+//crete a tcp session that will be use to send and receive data
+//
+//sesion sould be created succesfully and wait for commands
+func CreateSSHClient(WLclient SSHClient) (*ssh.Client, error) {
```

Comment:
We need to pass a [context](https://golang.org/pkg/context/) in here.  Instead of calling Dial we would then call (DialContext)[https://golang.org/pkg/net/#Dialer.DialContext].  This allows us to cancel and to timeout the SSH request.